### PR TITLE
feat(core): add LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeys

### PR DESCRIPTION
**Pull request by IceTDrinker**
### Description

- add spec and impls for 32/64 bits for fpksk required for a circuit
bootstrap

Introduces `LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeys` that will be used for circuit bootstrapping in the public API, the engine allowing to generate that entity is coming next.

Alternative name can be `LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeyVector`, but I'm starting to lose my mind over long names 😅 

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [ ] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: concrete-core/src/backends/default/implementation/entities/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+use crate::prelude::LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeysEntity;
+use crate::specification::entities::AbstractEntity;
+use concrete_commons::parameters::{
+    DecompositionBaseLog, DecompositionLevelCount, FunctionalPackingKeyswitchKeyCount,
+    GlweDimension, LweDimension,
+};
+#[cfg(feature = "backend_default_serialization")]
+use serde::{Deserialize, Serialize};
+
+/// A structure representing a private functional packing keyswitch key with 32 bits of precision.
```
Comment (agnesLeroy) [inline concrete-core/src/backends/default/implementation/entities/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys.rs:12]: a vector of private functional packing keyswitch keys

File: concrete-core/src/backends/default/implementation/entities/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+#[cfg(feature = "backend_default_serialization")]
+#[derive(Serialize, Deserialize)]
+pub(crate) enum LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeys32Version {
+    V0,
+    #[serde(other)]
+    Unsupported,
+}
+
+/// A structure representing a private functional packing keyswitch key with 64 bits of precision.
```
Comment (agnesLeroy) [inline concrete-core/src/backends/default/implementation/entities/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys.rs:56]: Same here

**Review by agnesLeroy — APPROVED** [review]

**Comment by agnesLeroy** [comment]
The name is ok for me, I agree adding Vector is too much, it's already very long.

**Comment by IceTDrinker** [comment]
mac OS build is stuck because of github actions it seems, merging

**Pull request merged**